### PR TITLE
Add Belarusian language

### PR DESCRIPTION
**Pull request by bespsm**
Pull request is work in progress.

There are many things that missing but my goal in this PR is to make sure that the "basic" phonemes generation works correctly based on:

https://star-wiki.ru/wiki/Belarusian_alphabet
https://en.m.wikipedia.org/wiki/Belarusian_phonology

and correctly mapped to Kirshenbaum according to:

https://en.wikipedia.org/wiki/Comparison_of_ASCII_encodings_of_the_International_Phonetic_Alphabet

7f7eee2c Belarusian: add initial dictionaries for phonemes, numbers.
723d4a98 Belarusian: add initial phonemes definition file.
9a7264d5 Belarusian: add initial language file.
afac2f1d Belarusian: add initial makefile rules

### CODE REVIEW

**Comment by valdisvi** [comment]
@bespsm, should I merge this pull request as is, or should I wait till you will notify additionally to do it?

**Comment by bespsm** [comment]
@valdisvi hey, thank you for asking, I'm just about to finish single letters spelling check and I after I think it will be ready to be merged. I will let you know.

File: dictsource/be_list [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ф       Ef
+х       xa
+ц       tsE
+ч       tS;E
+ш       s.a
+ы       y
+ь       m;ak;:i znak
+ю       ju
+я       ja
+'       ap;Ostraf
```
Comment (HaraldZealot) [inline dictsource/be_list:37]: Is semicolon here the sign of softness? If yes then why `p` is soft here?

File: dictsource/be_rules [hunk 2]
```diff
@@ -0,0 +1,110 @@
+// This file is UTF8 encoded
+// Translation rules for Belarusian
+
+// Special characters:
+// A  vowel:                   а е ё і о у ы э я
+// B  soft consonant:          ь й ч щ
```
Comment (HaraldZealot) [inline dictsource/be_rules]: Belarusian has no `щ`. And `ч` in Belarusian is always hard not soft (as in Russian)

File: dictsource/be_rules [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+// C  any consonant:           б в г д ж з й к л м н п р с т ў ф х ц ч ш ь ’
+// F  consonant, not hard:     б в г д з й к л м н п р с т ф х ч щ ь
```
Comment (HaraldZealot) [inline dictsource/be_rules]: `р` (эр) is only hard in Belarusian

File: dictsource/be_rules [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+// G  voiced:                  б в г д ж з
+// H  hard consonant:          ж ц ш
+// Y  iotated vowel, softsign: ь ю я ё
```
Comment (HaraldZealot) [inline dictsource/be_rules:11]: Where is `е`?

File: dictsource/be_rules [hunk 5]
```diff
@@ -0,0 +1,110 @@
+// This file is UTF8 encoded
+// Translation rules for Belarusian
+
+// Special characters:
+// A  vowel:                   а е ё і о у ы э я
+// B  soft consonant:          б п м ф в дз ц г к х 
+// C  any consonant:           б в г д ж з й к л м н п р с т ў ф х ц ч ш ь ’
```
Comment (HaraldZealot) [inline dictsource/be_rules:7]: I would say soft sign and especially apostrophe are not consonants

File: dictsource/be_rules [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+// F  consonant, not hard:     б в г д з й к л м н п р с т ф х ч  ь
```
Comment (HaraldZealot) [inline dictsource/be_rules:8]: `б`, `д`, `р`, `т`, `ч` can be only hard

File: dictsource/be_rules [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+// G  voiced:                  б в г д ж з
+// H  hard consonant:          ж ц ш
```
Comment (HaraldZealot) [inline dictsource/be_rules:10]: `б`, `д`, `р`, `т`, `ч` are missing
and is this category for "only hard" or "can be hard"?

File: dictsource/be_list (hunk 1 above)
Comment (bespsm) [reply dictsource/be_list:37]: thanks, it was copy-pasted from:

https://star-wiki.ru/wiki/Belarusian_alphabet

File: dictsource/be_rules (hunk 2 above)
Comment (bespsm) [reply dictsource/be_rules]: removed

File: dictsource/be_list (hunk 1 above)
Comment (bespsm) [reply dictsource/be_list]: removed

File: dictsource/be_rules (hunk 4 above)
Comment (bespsm) [reply dictsource/be_rules]: added

File: dictsource/be_rules [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+       ф       f
+
+.group х
+       х       x
+
+.group ц
+       ц       ts
+
+.group ч
+       ч       tS;    // FIXME: add ts.(tʂ) phoneme
```
Comment (HaraldZealot) [inline dictsource/be_rules:89]: should be tS it is always hard

File: dictsource/be_rules (hunk 3 above)
Comment (bespsm) [reply dictsource/be_rules]: removed

File: dictsource/be_rules (hunk 5 above)
Comment (bespsm) [reply dictsource/be_rules]: removed both

File: dictsource/be_rules (hunk 6 above)
Comment (bespsm) [reply dictsource/be_rules]: moved to hard

File: dictsource/be_list [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+п       pE
+р       Er
+с       Es
+т       tE
+у       u
+ў       u||n;Esklad;OvajE
+ф       Ef
+х       xa
+ц       tsE
+ч       tS;E
```
Comment (HaraldZealot) [inline dictsource/be_list:31]: tSE

File: dictsource/be_list [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+л       El
+м       Em
+н       En
+о       O
+п       pE
+р       Er
+с       Es
+т       tE
+у       u
+ў       u||n;Esklad;OvajE
```
Comment (HaraldZealot) [inline dictsource/be_list:27]: u||n;EskladOvajE

File: dictsource/be_list [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+б       bE
+в       vE
+г       QE
+д       dE
+е       jE
+ё       j;O
+ж       z.E
+з       zE
+і       i
+й       i||n;Esklad;OvajE
```
Comment (HaraldZealot) [inline dictsource/be_list:16]: i||n;EskladOvajE

File: dictsource/be_list [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+
+// Letter names
+а       a
+_б      bE
+б       bE
+в       vE
+г       QE
+д       dE
+е       jE
+ё       j;O
```
Comment (HaraldZealot) [inline dictsource/be_list:12]: jO

File: dictsource/be_rules (hunk 7 above)
Comment (bespsm) [reply dictsource/be_rules]: I guess it includes both categories, "only hard" and "can be hard"

File: dictsource/be_list [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+у       u
+ў       u||n;Esklad;OvajE
+ф       Ef
+х       xa
+ц       tsE
+ч       tS;E
+ш       s.a
+ы       i"
+ь       m;ak;:i||znak
+ю       ju
```
Comment (HaraldZealot) [inline dictsource/be_list:35]: you forget "э"

File: dictsource/be_rules (hunk 7 above)
Comment (bespsm) [reply dictsource/be_rules]: added the letters

File: dictsource/be_rules [hunk 14]
```diff
@@ -0,0 +1,110 @@
+// This file is UTF8 encoded
+// Translation rules for Belarusian
+
+// Special characters:
+// A  vowel:                   а е ё і о у ы э я
+// B  soft consonant:          б п м ф в дз ц г к х
```
Comment (HaraldZealot) [inline dictsource/be_rules:6]: `з`, `л`, `н`, `с`

Comment (HaraldZealot) [reply dictsource/be_rules:6]: or it is "only soft"?

File: dictsource/be_list (hunk 10 above)
Comment (bespsm) [reply dictsource/be_list]: fixed

File: dictsource/be_list (hunk 11 above)
Comment (bespsm) [reply dictsource/be_list]: fixed

File: dictsource/be_list (hunk 13 above)
Comment (bespsm) [reply dictsource/be_list:35]: shame on me, thank you

File: dictsource/be_rules (hunk 14 above)
Comment (bespsm) [reply dictsource/be_rules]: added

File: dictsource/be_list (hunk 12 above)
Comment (bespsm) [reply dictsource/be_list]: fixed

File: dictsource/be_list [hunk 15]
```diff
@@ -0,0 +1,78 @@
+// This file is UTF8 encoded
+// Spelling-to-phoneme words for Belarusian
+
+// Letter names
+а       a
+_б      bE
```
Comment (HaraldZealot) [inline dictsource/be_list:6]: why you keep `_б` between alphabetical letters?

Comment (bespsm) [reply dictsource/be_list]: there is a meaning definitely, but I don't know it yet. I remove it for now

File: dictsource/be_rules [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+       ы       i"    // (ɨ
+
+.group ь
+       ь       ;    // ʲ
+
+.group э
+       э       E    // ɛ
+
+.group ю
+       ю       ju
```
Comment (HaraldZealot) [inline dictsource/be_rules:104]: why `ё` is `;O` but `ю` is `ju`?

File: dictsource/be_rules [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+
+.group я
+       я       ja
```
Comment (HaraldZealot) [inline dictsource/be_rules]: same here

File: dictsource/be_rules [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+
+.group '
```
Comment (HaraldZealot) [inline dictsource/be_rules:109]: I'm not sure that you can solve apostrophe just by substitute it by `j`

File: dictsource/be_rules [hunk 19]
```diff
@@ -0,0 +1,110 @@
+// This file is UTF8 encoded
+// Translation rules for Belarusian
+
+// Special characters:
```
Comment (HaraldZealot) [inline dictsource/be_rules:4]: in the table `дз` and `дж` is totally missed

File: dictsource/be_rules (hunk 16 above)
Comment (bespsm) [reply dictsource/be_rules]: fixed

File: dictsource/be_list (hunk 13 above)
Comment (bespsm) [reply dictsource/be_list]: added

File: dictsource/be_rules (hunk 8 above)
Comment (bespsm) [reply dictsource/be_rules]: fixed

File: dictsource/be_list (hunk 9 above)
Comment (bespsm) [reply dictsource/be_list]: fixed

File: dictsource/be_rules (hunk 17 above)
Comment (bespsm) [reply dictsource/be_rules]: fixed

File: dictsource/be_rules (hunk 18 above)
Comment (bespsm) [reply dictsource/be_rules:109]: for the first iteration I think it's fine. here is output from helper service with з'ем:

https://baltoslav.eu/ipa/index.php?mova=en&j=by&t=%D0%B7%27%D0%B5%D0%BC

here is my output, and actually it sounds very close:

`Translate 'з'ем'
 36     з       [z]

  1     '        [j]

 36     е       [;E]

 36     м       [m]

zjˈɛm`

File: dictsource/be_rules (hunk 19 above)
Comment (bespsm) [reply dictsource/be_rules:4]: de facto it's not the charecters. But later on we need to add the rules for them. In this PR I want to implement only correct single character mapping

File: dictsource/be_rules (hunk 17 above)
Comment (bespsm) [reply dictsource/be_rules]: fixed

File: dictsource/be_rules (hunk 8 above)
Comment (bespsm) [reply dictsource/be_rules]: fixed

File: dictsource/be_rules (hunk 18 above)
Comment (HaraldZealot) [reply dictsource/be_rules:109]: So you creating palatalized `j`????

File: dictsource/be_rules (hunk 19 above)
Comment (HaraldZealot) [reply dictsource/be_rules:4]: There is a big question how the system (for which you create dictionaries) works. I need to know a bit more about it, however it is very important to note:
almost in any language `character != sound`
There is for example in English one character which refers to cluster sound (diphthong, i.e. literally double sound from Greek, however it is more complex unity from modern phonology POV). Namely letter/character `i` (like personal pronoun 1st person singular) is `[aɪ]`. The same for Cyrillic isolated character `я` it means two sounds `[ja]`. And in the opposite way there are character cluster which means one sound like digraphs `th` is used for one sound `[θ]`, or another digraph `sh` is one sound `[ʃ]` (in German the trigraph `sch` is used, and Cyrillic has just one letter `ш` for the same sound). So the same are `дз` and `дж`, i.e. digraphs.

The problem with this helping table it is misleading character is never consonant or vowel. The SOUND is consonant or vowel.

File: dictsource/be_rules (hunk 18 above)
Comment (bespsm) [reply dictsource/be_rules:100]: `ˈ` it's stress

**Comment by HaraldZealot** [comment]
LGTM

File: dictsource/be_rules (hunk 19 above)
Comment (bespsm) [reply dictsource/be_rules]: removed comment section

File: dictsource/be_rules (hunk 8 above)
Comment (HaraldZealot) [reply dictsource/be_rules]: the very last one we already has add ts. so let use it

Comment (bespsm) [reply dictsource/be_rules]: it's outdated comment, if you check `ч` it's already mapped to `ts.`

**Comment by bespsm** [comment]
@valdisvi review is finish, so please check it

**Comment by valdisvi** [comment]
Thanks!
I included proposed changes as one related commit 7aacdef65d9ec56f6620a8335851e32fb31708f5 with some additions for documentation and tests.

**Pull request closed without merging**